addons/purchase/tests/test_purchase_order_report.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from datetime import datetime
+from odoo.tests import common
+
+
+@common.tagged('post_install','-at_install')
+class TestPurchaseOrderReport(common.TransactionCase):
+    def setUp(self):
+        super(TestPurchaseOrderReport, self).setUp()
+
+        self.partner_id = self.env.ref('base.res_partner_1')
+        self.product1 = self.env.ref('product.product_product_8')
+        self.product2 = self.env.ref('product.product_product_11')
+        self.PurchaseReport = self.env['purchase.report']
+
+        # Create a new company and set CoA
+        self.env.user.company_id = self.env['res.company'].create({'name': 'new_company'})
+        self.env.ref('l10n_generic_coa.configurable_chart_template').load_for_current_company(False, False)
+
+    def test_00_purchase_order_report(self):
+        uom_dozen = self.env.ref('uom.product_uom_dozen')
+
+        eur_currency = self.env.ref('base.EUR')
+        self.cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", [self.env.ref('base.USD').id, self.env.company_id.id])
+        self.env['res.currency.rate'].search([]).unlink()
+        self.env['res.currency.rate'].create({
+            'name': datetime.today(),
+            'rate': 2.0,
+            'currency_id': eur_currency.id,
+        })
+        po1 = self.env['purchase.order'].create({
+            'partner_id': self.partner_id.id,
+            'currency_id': eur_currency.id,
+            'order_line': [
+                (0, 0, {
+                    'name': self.product1.name,
+                    'product_id': self.product1.id,
+                    'product_qty': 1.0,
+                    'product_uom': uom_dozen.id,
+                    'price_unit': 100.0,
+                    'date_planned': datetime.today(),
+                }),
+                (0, 0, {
+                    'name': self.product2.name,
+                    'product_id': self.product2.id,
+                    'product_qty': 1.0,
+                    'product_uom': uom_dozen.id,
+                    'price_unit': 200.0,
+                    'date_planned': datetime.today(),
+                }),
+            ],
+        })
+        po1.button_confirm()
+
+        invoice_po1 = self.env['account.invoice'].create({
```

Comment:
Use a Form() in order to avoid manually onchange trigger.

Reply:
@amoyaux  done